File: software/examples/python/example_touch.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change XYZ to the UID of your LCD 128x64 Bricklet

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_lcd_128x64 import BrickletLCD128x64

# Callback function for touch_position callback
def cb_touch_position(pressure, x, y, age):
    print("Pressure: " + str(pressure))
    print("X: " + str(x))
    print("Y: " + str(y))
    print("Age: " + str(age))
    print("")

# Callback function for touch_gesture callback
def cb_touch_gesture(gesture, duration, x_start, x_end, y_start, y_end, age):
    print("Gesture: " + str(gesture))
    print("Duration: " + str(duration))
    print("X Start: " + str(x_start))
    print("X End: " + str(x_end))
    print("Y Start: " + str(y_start))
    print("Y End: " + str(y_end))
    print("Age: " + str(age))
    print("")

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    lcd = BrickletLCD128x64(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Register touch_position callback to function cb_touch_position
    lcd.register_callback(lcd.CALLBACK_TOUCH_POSITION, cb_touch_position)

    # Register touch_gesture callback to function cb_touch_gesture
    lcd.register_callback(lcd.CALLBACK_TOUCH_GESTURE, cb_touch_gesture)

    # Configure touch position callback with a period of 100ms
    lcd.set_touch_position_callback_configuration(100, True)

    # Configure touch gesture callback with a period of 100ms
    lcd.set_touch_gesture_callback_configuration(100, True)

    raw_input("Press key to exit\n") # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleTouch.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your LCD 128x64 Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var lcd = new Tinkerforge.BrickletLCD128x64(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Set period for touch position callback to 0.1s (100ms)
        lcd.setTouchPositionCallbackConfiguration(100, true);

        // Set period for touch gesture callback to 0.1s (100ms)
        lcd.setTouchGestureCallbackConfiguration(100, true);
    }
);

// Register touch position callback
lcd.on(Tinkerforge.BrickletLCD128x64.CALLBACK_TOUCH_POSITION,
    // Callback function for touch position callback
    function (pressure, x, y, age) {
        console.log('Pressure: ' + pressure);
        console.log('X: ' + x);
        console.log('Y: ' + y);
        console.log('Age: ' + age);
        console.log();
    }
);

// Register touch gesture callback
lcd.on(Tinkerforge.BrickletLCD128x64.CALLBACK_TOUCH_GESTURE,
    // Callback function for touch gesture callback
    function (gesture, duration, xStart, xEnd, yStart, yEnd, age) {
        console.log('Gesture: ' + gesture);
        console.log('Duration: ' + duration);
        console.log('X Start: ' + xStart);
        console.log('X End: ' + xEnd);
        console.log('Y Start: ' + yStart);
        console.log('Y End: ' + yEnd);
        console.log('Age: ' + age);
        console.log();
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
